### ISSUE TITLE
Raspberry Pi Zero W (GPi Case): Black screen on start

### ISSUE THREAD

**Issue by Bright5park** [sm64pc/sm64pc#27]
After some trial and error on other platforms I have managed to compile it straight on my Raspberry Pi Zero running Retropie/Raspbian (which took a good while), however it does not show any graphics. Input and audio seem to work fine, but no video. The console also doesn't note any errors:

pi@retropie:~ $ ./RetroPie/roms/ports/sm64/sm64
Loading configuration from 'sm64config.txt'
option: 'fullscreen', value: 'true'
option: 'key_a', value: '38'
option: 'key_b', value: '51'
option: 'key_start', value: '57'
option: 'key_r', value: '54'
option: 'key_z', value: '37'
option: 'key_cup', value: '328'
option: 'key_cdown', value: '336'
option: 'key_cleft', value: '331'
option: 'key_cright', value: '333'
option: 'key_stickup', value: '17'
option: 'key_stickdown', value: '31'
option: 'key_stickleft', value: '30'
option: 'key_stickright', value: '32'
^CSaving configuration to 'sm64config.txt'
pi@retropie:~ $

One thing I had to change to get it to compile was to modify the Makefile, because gcc complained that it couldn't find/handle the march "armv6zk+fp", so I modified it to replace the offending line

                OPT_FLAGS := -march=armv6zk+fp -mfpu=vfp -Ofast

with

                OPT_FLAGS := -march=armv6zk -mfpu=vfp -Ofast

Also (because I copied it over from Windows via WinSCP, I guess) I had to recursively make the entire project folder executable and take ownership of it

Otherwise I obeyed the compilation guide to the letter; installed the dependencies and ran "make TARGET_RPI=1"

**Comment by Hyenadae** [comment]
I compile on a Pi 1 B+ (512MB) since the start and never had any issues with output (vc4_drm and Mesa 19.2.1 OGL renderer)

What version is your GCC / Raspbian install? Can you try a fresh latest Raspbian SD, install/enable accelerated video + test it out?

**Comment by Hyenadae** [comment]
Please see : 
https://github.com/sm64pc/sm64pc/issues/37
https://github.com/sm64pc/sm64pc/issues/35 

I think I've gotten this all figured out and provided solutions :)

**Comment by Bright5park** [comment]
Seems like the GPi had the same idea about reinstalling Retropie, as it didn't feel like shutting down last night, so I forced it to shut down, and I just returned to it to see that that made it mangle it's partition. I am able to recover the data with my PC, but it's no longer bootable - D'oh!

Oh well, once I get her back up and running again I'll go through the two posts and see if I can make SM64 sing - or rather, give me visuals while it sings. Thanks!

**Comment by Hyenadae** [comment]
Good luck! Keep in mind that the GPi's screen is likely using the SPI? interface, which is a separate device than vc4_drm (/dev/card1 or 0). X11/emustation is probably configured with a special XRandr option to do the rotations (/etc/X11/xorg.conf.d ?) which doesn't apply to VC4 as it's a different driver, and may use a different conf file

**Comment by Bright5park** [comment]
Okay, I think I have run into a significant snag:

Generally, running Super Mario 64 on the Pi Zero is possible, but I feel not in combination with the GPi case, because - from what information I could find online - the OpenGL driver for the Pi does apparently not support video out via GPIO.

Technically, there is a work around by using a secondary framebuffer and outputting that to GPIO, but the problem is that the Pi Zero is (partially thanks to the GPi case that seems to be designed to run the Pi Zero as hot as possible) already reaching it's limits rather quickly, which to me smells like loss of playability if I introduce the framebuffer in certain emulators that are already just about pushing the Zero to it's limits.

And to be honest, the available controls of the GPi case are insufficient (only SNES-tier inputs, so no analog inputs and only four face buttons and two shoulder buttons), so I would have to either do some really funky rebinding or connect a Bluetooth gamepad, and frankly: At that point I might as well use my phone.

However: The guide at #37 worked perfectly, although I ran into memory issues a few times during compilation until I removed the OpenGL-specific memory settings again, so I think that the port should be compiled first, and then the Pi Zero can be switched to OpenGL mode.

So, in conclusion: It's not the port, it's the case that's the problem. If you have a portable shell that uses HDMI to hook up to a display (or connect it to a monitor) then it will work fine.

Thanks a bunch for the help!
